core/src/display_object/movie_clip.rs

```diff
@@ -421,8 +419,8 @@ impl<'gc> MovieClip<'gc> {
 
     /// Execute all other timeline actions on this object.
     pub fn run_frame_avm1(self, context: &mut UpdateContext<'gc>) {
-        if !self.movie().is_action_script_3() {
-            // Run my load/enterFrame clip event.
+        // Run my load/enterFrame clip event.
+        if !self.movie().is_action_script_3() || !context.root_swf.is_action_script_3() {
```

Comment:
Could you also add a comment here explaining this situation? I.e. why it is being run for an AVM2 movie